### PR TITLE
CJ4: Fix SimBrief handling of three letter airports

### PR DESCRIPTION
**Pull request by dga711**
SimBrief sends for ex. 3U2 as K3U2 and the sim doesnt like it.

If there is a number after K, it will strip the K and add a space at the end so the sim takes it.

### CODE REVIEW

File: src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js [hunk 1]
```diff
@@ -3,6 +3,14 @@ class CJ4_FMC_FplnRecallPage {
         let url = "https://www.simbrief.com/api/xml.fetcher.php?userid=" + pilotId + "&json=1";
         let json = "";
 
+        let parseAirport = (icao) => {
+            if (icao.startsWith("K") && (/\d/.test(icao))) {
```
Comment (Jessidhia) [inline src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]:
```suggestion
            if (/^K\d/.test(icao)) {
```

Comment (mattnischan) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: I'm also confused on that. So if it's `KH2L`, we want to strip out the K also?

Comment (dga711) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: > I'm also confused on that. So if it's `KH2L`, we want to strip out the K also?

yes. the game wants "H2L "  it doesnt accept KH2L

Comment (dga711) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: > Actually, I'm not sure what you meant by "If there is a number after K". If it starts with K and there's a number anywhere, or literally if there's a number "after K"?

Thanks for the suggestion of doing it smarter. I will gladly take it ;)

Comment (dga711) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: So. The number can be anywhere in the ident after the K. So the test as suggested wouldn't work.
Needs optional other characters in between

Comment (dga711) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: How about 
image

Comment (mattnischan) [reply src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_FplnRecallPage.js:7]: Was just going to suggest that, yep. Looks good.

**Review by cwburnett — APPROVED** [review]

**Pull request merged**